File: app/src/main/java/com/example/listapp/di/module/NetworkModule.java
package com.example.listapp.di.module;

import com.example.listapp.model.retrofit.RetrofitClient;
import com.example.listapp.model.retrofit.RetrofitServices;

import dagger.Module;
import dagger.Provides;

@Module
public class NetworkModule {
    public static final String BASE_URL = "https://pastebin.com/raw/";

    @Provides
    public RetrofitServices provideRetrofitServices() {
        return RetrofitClient.INSTANCE
                .getClient(BASE_URL)
                .create(RetrofitServices.class);
    }
}

File: app/src/main/java/com/example/listapp/di/module/NetworkModule.kt
package com.example.listapp.di.module

import com.example.listapp.model.retrofit.RetrofitClient
import com.example.listapp.model.retrofit.RetrofitServices

class NetworkModule {

    fun provideRetrofitServices(): RetrofitServices {

        return RetrofitClient
            .getClient(BASE_URL)
            .create(RetrofitServices::class.java)
    }

    private companion object {
        const val BASE_URL = "https://pastebin.com/raw/"
    }
}